extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/context/ArcContextProvider.java

```diff
@@ -49,17 +41,31 @@ public ThreadContextSnapshot currentContext(Map<String, String> map) {
                 // context active, store current state, feed it new one and restore state afterwards
                 InjectableContext.ContextState stateToRestore = requestContext.getState();
```

Comment:
This could throw `ContextNotActiveException`...

Reply:
This only gets called if the above condition holds true which should make sure you don't get that exception.